libs/server/Resp/AdminCommands.cs

```diff
@@ -250,33 +250,102 @@ private bool ProcessAdminCommands<TGarnetApi>(RespCommand command, ReadOnlySpan<
             }
             else if (command == RespCommand.COMMAND)
             {
-                if (count != 0)
+                var subCommand = ReadOnlySpan<byte>.Empty;
+
+                if (count > 0)
                 {
-                    if (!DrainCommands(bufSpan, count))
+                    subCommand = GetCommand(bufSpan, out var success);
+                    if (!success)
                         return false;
-                    errorFlag = true;
-                    errorCmd = "command";
                 }
-                else
+
+                // Handle COMMAND COUNT
+                if (count > 0 && (subCommand.SequenceEqual(CmdStrings.COUNT) || subCommand.SequenceEqual(CmdStrings.count)))
                 {
-                    // TODO: include the built-in commands
-                    string resultStr = "";
-                    int cnt = 0;
-                    for (int i = 0; i < storeWrapper.customCommandManager.CommandId; i++)
+                    if (!RespCommandsInfo.TryGetRespCommandsInfoCount(out var respCommandCount, true, logger))
                     {
-                        var cmd = storeWrapper.customCommandManager.commandMap[i];
-                        if (cmd != null)
+                        respCommandCount = 0;
+                    }
+
+                    var commandCount = storeWrapper.customCommandManager.CustomCommandsInfoCount + respCommandCount;
+
+                    while (!RespWriteUtils.WriteInteger(commandCount, ref dcurr, dend))
+                        SendAndReset();
+                }
+                // Handle COMMAND and COMMAND INFO
+                else if (count == 0 || subCommand.SequenceEqual(CmdStrings.INFO) || subCommand.SequenceEqual(CmdStrings.info))
```

Comment:
I know this is not specific to this PR, but I think `COMMAND INFO` shouldn't require admin permissions.

Reply:
Good catch!